test/functional/InvalidArgumentExceptionTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Dhii\Exception\FuncTest;
+
+use Xpmock\TestCase;
+use Exception as RootException;
+
+/**
+ * Tests {@see \Dhii\Exception\Exception}.
+ *
+ * @since [*next-version*]
+ */
+class InvalidArgumentExceptionTest extends TestCase
+{
+    /**
+     * The name of the test subject.
+     *
+     * @since [*next-version*]
+     */
+    const TEST_SUBJECT_CLASSNAME = 'Dhii\\Exception\\InvalidArgumentException';
+
+    /**
+     * Creates a new instance of the test subject.
+     *
+     * @since [*next-version*]
+     *
+     * @return \Dhii\Exception\Exception
+     */
+    public function createInstance($message = '', $code = 0, $previous = null, $argument = null)
+    {
+        $mock = $this->mock(static::TEST_SUBJECT_CLASSNAME)
+                ->new($message, $code, $previous, $argument);
+
+        return $mock;
+    }
+
+    /**
+     * Creates a new exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string             $message  The message.
+     * @param int                $code     The code.
+     * @param RootException|null $previous The inner exception, if any.
+     *
+     * @return RootException The new exception.
+     */
+    public function createException($message = '', $code = 0, $previous = null)
+    {
+        return new RootException($message, $code, $previous);
+    }
+
+    /**
+     * Tests whether a valid instance of the test subject can be created.
+     *
+     * @since [*next-version*]
+     */
+    public function testCanBeCreated()
+    {
+        $subject = $this->createInstance();
+
+        $this->assertInstanceOf(static::TEST_SUBJECT_CLASSNAME, $subject, 'Subject is not a valid instance.');
+    }
+
+    /**
+     * Tests that the exception params can be correctly set in the constructor,
+     * and can be correctly retrieved.
+     *
+     * @since [*next-version*]
+     */
+    public function testConstructorAndGetters()
+    {
+        $message = uniqid('message-');
+        $code = rand(1, 100);
+        $previous = $this->createException(uniqid('inner-message-'), rand(1, 100));
+        $argument = uniqid('argument-');
+        $subject = $this->createInstance($message, $code, $previous, $argument);
+
+        $this->assertInstanceOf('Exception', $subject, 'Subject is not a valid exception');
+        try {
+            throw $subject;
+        } catch (RootException $e) {
+            $this->assertInstanceOf(static::TEST_SUBJECT_CLASSNAME, $subject, 'A valid instance of the test subject could not be created');
+            $this->assertInstanceOf('Dhii\\Exception\\InvalidArgumentExceptionInterface', $subject, 'Subject is not a valid Dhii throwable');
+            $this->assertEquals($message, $e->getMessage(), 'Subject message is wrong');
```

Comment:
I thought about that. But on the other hand, this is a different concrete class, with a different constructor, which sets the internal properties, and therefore needs to be tested separately.